src/careamics/dataset/in_memory_dataset.py

```
@@ -218,13 +218,7 @@ def __getitem__(self, index: int) -> tuple[np.ndarray, ...]:
 
             return self.patch_transform(patch=patch, target=target)
 
-        elif isinstance(self.data_config, N2VDataConfig):
-            return self.patch_transform(patch=patch)
-        else:
-            raise ValueError(
-                "Something went wrong! No target provided (not supervised training) "
-                "while the algorithm is not Noise2Void."
-            )
+        return self.patch_transform(patch=patch)
```

I think we still want to raise an error here if the user does not provide a target for a supervised algorithm

---

For that we'd need to rename those and in general this will be redone in the dataset refac

---

I agree we should fail early when providing targets to N2V.

I guess that the problem now is that the dataset has no way to know whether it is Noise2Void. Which is an issue, the next time it can fail is then in the Lightning module.

We didn't catch that in the previous PR. So it should probably be covered in the `CAREamist` class, which has the correct knowledge and can check the target in the `careamist.train` call.

> For that we'd need to rename those and in general this will be redone in the dataset refac

I don't understand what you mean.

---

I'll think how to do it better